lib/instrumentation/modules/memcached.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+var semver = require('semver')
+
+var shimmer = require('../shimmer')
+
+module.exports = function (memcached, agent, { version, enabled }) {
+  if (!enabled) {
+    return memcached
+  }
+  if (!semver.satisfies(version, '>=2.2.2')) {
+    agent.logger.debug('Memcached version %s not supported - aborting...', version)
+    return memcached
+  }
+  agent.logger.debug('shimming memcached.prototype.command')
+  shimmer.wrap(memcached.prototype, 'command', wrapCommand)
+  return memcached
+
+  // Wrap the generic command that is used to build touch, get, gets etc
+  function wrapCommand (original) {
+    return function wrappedCommand () {
+      if (typeof arguments[0] === 'function') {
+        var query = arguments[0]()
+        // If the callback is not a function the user doesn't care about result
+        if (typeof query.callback === 'function') {
+          var span = agent.startSpan(`memcached.${query.type}`)
```

Comment:
I'm wondering if this should include the key as well. 
It makes it look like:
![image](https://user-images.githubusercontent.com/30619604/60333121-8bbcdd80-9998-11e9-9132-95df89592394.png)
Open to suggestions if its better to add it as a label etc...

Reply:
Probably better as a label. @watson Do you agree?

Reply:
@Qard I had the page open in GitHub while doing the review, so didn't see your comment until now. A label is fine as well, but you can't see it at a glance. So if this is relevant information on the waterfall screen, then it's probably better to have it in the name. Is there any downside to showing it in the name?

Reply:
Depends. With SQL we have the summary which removes unique data and just retains the query structure, allowing similar queries to overlap, here we don't have that which could result in an explosion of unique names. Would that matter much for fingerprinting, or do you think just going by the type is enough? 🤔

Reply:
I was thinking it was kind of like including the table name as we do in SQL - but I guess the analogy doesn't quite hold. Let's leave it out of the name and just make sure to include it using `span.setDbContext()` instead.

I was thinking we should include the query type in there as well so that it will be `get foo` instead of just `foo`. Or `getMulti foo,bar` instead of just `foo,bar`:

```js
span.setDbContext({ statement: `${query.type} ${query.key}`, type: 'memcached' })
```